ratelimitj-redis/src/main/resources/sliding-window-ratelimit.lua

```diff
@@ -75,11 +103,12 @@ if weight ~= 0 then
     end
 end
 
--- We calculated the longest-duration limit so we can EXPIRE
+-- We calculated the longest-duration and longest-applied-backoff limit so we can EXPIRE
 -- the whole HASH for quick and easy idle-time cleanup :)
-if longest_duration > 0 then
+local longest_expire = math.max(longest_applied_backoff, longest_duration)
```

Comment:
We can't expire the key on the duration, if we have a applied backoff that is longer than the duration.